lib/src/migrators/division.dart

```diff
@@ -0,0 +1,208 @@
+
+import 'package:args/args.dart';
+
+// The sass package's API is not necessarily stable. It is being imported with
+// the Sass team's explicit knowledge and approval. See
+// https://github.com/sass/dart-sass/issues/236.
+import 'package:sass/src/ast/sass.dart';
+
+import 'package:sass_migrator/src/migration_visitor.dart';
+import 'package:sass_migrator/src/migrator.dart';
+import 'package:sass_migrator/src/patch.dart';
+import 'package:sass_migrator/src/utils.dart';
+
+/// Migrates stylesheets that use the `/` operator for division to use the
+/// `divide` function instead.
+class DivisionMigrator extends Migrator {
+  final name = "division";
+  final description =
+      "Migrates from the / division operator to the divide function";
+
+  @override
+  final argParser = ArgParser()
+    ..addFlag('aggressive',
+        abbr: 'a',
+        help: r"If set, expressions like '$x/3 + 1', '$x/3 - 1', and 'fn()/3' "
+            "will be migrated.");
+
+  bool get isAggressive => argResults['aggressive'] as bool;
+
+  Map<Uri, String> migrateFile(Uri entrypoint) =>
+      _DivisionMigrationVisitor(this.isAggressive, migrateDependencies)
+          .run(entrypoint);
+}
+
+class _DivisionMigrationVisitor extends MigrationVisitor {
+  final bool isAggressive;
+
+  _DivisionMigrationVisitor(this.isAggressive, bool migrateDependencies)
+      : super(migrateDependencies: migrateDependencies);
+
+  /// True when division is allowed by the context the current node is in.
+  bool _isDivisionAllowed = false;
+
+  /// True when the current node is expected to evaluate to a number.
+  bool _expectsNumericResult = false;
+
+  /// If this is a division operation, migrates it.
+  ///
+  /// If this is any other operator, allows division within its left and right
+  /// operands.
+  @override
+  void visitBinaryOperationExpression(BinaryOperationExpression node) {
+    if (node.operator == BinaryOperator.dividedBy) {
+      if (shouldMigrate(node)) {
+        addPatch(patchBefore(node, "divide("));
+        patchSlashToComma(node);
+        addPatch(patchAfter(node, ")"));
+      }
+      super.visitBinaryOperationExpression(node);
+    } else {
+      withContext(
+          true,
+          _expectsNumericResult || operatesOnNumbers(node.operator),
+          () => super.visitBinaryOperationExpression(node));
+    }
+  }
+
+  /// Disallows division within this list.
+  @override
+  void visitListExpression(ListExpression node) {
+    withContext(
+        false, _expectsNumericResult, () => super.visitListExpression(node));
+  }
+
+  /// If this parenthesized expression contains a division operation, migrates
+  /// it using the parentheses that already exist.
+  @override
+  void visitParenthesizedExpression(ParenthesizedExpression node) {
+    var expression = node.expression;
+    if (expression is BinaryOperationExpression &&
+        expression.operator == BinaryOperator.dividedBy) {
+      withContext(true, _expectsNumericResult, () {
```

Comment:
You can avoid some repetition and extra nesting by bringing `withContext()` to the root of the method:

```dart
void visitParenthesizedExpression(ParenthesizedExpression node) {
  withContext(true, _expectsNumericResult, () {
    var expression = node.expression;
    if (expression is BinaryOperationExpression &&
        expression.operator == BinaryOperator.dividedBy &&
        shouldMigrate(expression)) {
      addPatch(patchBefore(node, "divide"));
      patchSlashToComma(expression);
      super.visitBinaryOperationExpression(expression);
    } else {
      super.visitParenthesizedExpression(node);
    }
  });
}
```

Reply:
I moved `_withContext` to the root of the method, but left the `_shouldMigrate` check as its own if statement, since combining it in this way will cause it to be called a second time on the same expression in `visitBinaryOperationExpression`, which would output duplicate warnings.